add predicate filter option

demmer opened the pull request:
Make it possible to retry only upon errors that are eligible by the predicate option.

Adapted from the original proposal by @gastonelhordoy in #27

gastonelhordoy commented on lib/bluebird-retry.js:
```diff
@@ -65,6 +65,9 @@ function retry(func, options) {
                         return Promise.reject(err);
                     }
                 }
+                if (options.predicate && !(err instanceof options.predicate)) {
```
I think it shouldn't be negated, should it?

demmer commented:
@gastonelhordoy You made a good point in #27 regarding predicate functions.

I reworked the implementation so it always uses bluebird's filtered catch, supplying a default predicate that always succeeds if the user isn't using one. I also updated your tests slightly to verify it works if the user supplies either an error class name or a predicate function.

gastonelhordoy commented:
Thanks @demmer! I think it's pretty much what I was looking for. Are you gonna releasing a new version with this?

demmer commented:
@gastonelhordoy Done: https://github.com/demmer/bluebird-retry/releases/tag/v0.7.0
